### DISCUSSION TITLE
pos_ydir vs. reference coordinates in GUI

### DISCUSSION THREAD

**Discussion by ntardiff** [simnibs/simnibs#444] [8. Simulation workflows]
I'm trying to switch from using simnibs_gui to scripting for simulating a TMS efield. I tried using the same coordinates I have been using for the reference coordinates in GUI for pos_ydir in a matlab script, with the same coil center. However, I find that this gives me different results, and when I visualize the coil direction in gmsh it is oriented differently. The descriptions of pos_ydir and reference coordinates sound very similar in the documentation, so I must be missing something re: how they differ? Is there a formula I can use to switch from one to the other?

Thanks,
Nathan

**Comment by ntardiff** [comment]
Update: 

In my original calculations, I only translated the ref coordinate one unit along the y-axis. I found that when I translate it 20 units instead (as recommended in some of the other threads here e.g. https://github.com/simnibs/simnibs/discussions/318, I get a coil orientation that is consistent (at least qualitatively) using matlab scripting. 

Perhaps something about the orthogonalization in calc_matsimnibs is messing things up when they are too close together? But I don't understand why that wouldn't happen w/ the GUI interface?

Thanks again,
Nathan

**Reply by TorgeW** [reply]
Are you using project to scalp in the GUI? In the script this is done automatically and might be the reason for the different positions.

**Reply by ntardiff** [reply]
From limited testing, I found that whether or not you click project to scalp, the resulting simulation (and associated matsimnibs matrix) are the same, so I think it's projecting to scalp and then adding the coil distance from scalp anyway? Though I could be wrong.

Anyway, like I said I resolved the discrepancy by translating pos_ydir a larger distance in the scripting environment, so it's no longer a problem, just a curiosity...
